bundles/org.openhab.binding.ephemeris/src/main/java/org/openhab/binding/ephemeris/internal/handler/HolidayHandler.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ephemeris.internal.handler;
+
+import static org.openhab.binding.ephemeris.internal.EphemerisBindingConstants.*;
+
+import java.time.*;
+import java.util.*;
+
+import org.eclipse.jdt.annotation.*;
+import org.openhab.binding.ephemeris.internal.providers.EphemerisDescriptionProvider;
+import org.openhab.core.ephemeris.EphemerisManager;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.*;
+import org.openhab.core.types.StateOption;
+
+/**
+ * The {@link HolidayHandler} delivers system default Holidays data.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class HolidayHandler extends JollydayHandler {
+    private final List<StateOption> events = new ArrayList<>();
+
+    public HolidayHandler(Thing thing, EphemerisManager ephemerisManager, ZoneId zoneId,
+            EphemerisDescriptionProvider descriptionProvider) {
+        super(thing, ephemerisManager, zoneId);
+
+        // Search all holidays in the coming year
+        ZonedDateTime now = ZonedDateTime.now();
+        for (int offset = 0; offset < 366; offset++) {
+            String nextEvent = getEvent(now.plusDays(offset));
+            if (nextEvent != null) {
+                String description = ephemeris.getHolidayDescription(nextEvent);
+                events.add(new StateOption(nextEvent, description == null ? nextEvent : description));
```

Comment:
No need to refresh this table every day ?
If not, every day passing, you will miss potentially one coming event.

Reply:
I will not miss any events. This loop is only here to initialize values of the descriptionProvider for a full year.

Reply:
By going from toady until today+365, is there no risk that you take twice the same holiday and so create two state options for the same holiday ?
Imagine today is Christmas day, today+365 will also be a Christmas day.
Stopping at today+364 should avoid the case of the holiday at a fixed date.

Here is another problem for holiday that are at different data depending on the year. For example, the "lundi de Pâques". In 2024, it was the first of April. In 2025, it will be the 21th of April. Imagine you start the binding the 8th of April 2024 for example, you will not get the proper name of "lundi de Pâques" of 2025.

As I understand, this is only to get the proper translation of the holiday, so it is not dramatic.

In my opinion, the best approach will be to consider the next 13 months for examples and to handle duplicates to avoid creating duplicated state options. WDYT ?

Reply:
At the same time, the probability to not restart openHAB during one year is low ;) So maybe my second point could be "ignored".
My first remark remains valid (maybe stop at today+364 to avoid duplicated options).

Reply:
Maybe you could create a map before finally converting it to a List. This will avoid the duplicates.

Reply:
Now scanning 13 montes and using a map.

Reply:
Well done